From b32d6b162899dacd8e833dcf58e193dd57091858 Mon Sep 17 00:00:00 2001
From: Damian Wrobel <dwrobel@ertelnet.rybnik.pl>
Date: Tue, 31 May 2022 20:13:37 +0200
Subject: [PATCH] Revert "cgroup: Disable cgroup "memory" by default"

This reverts commit 16f8c6bc6e901dcf4402e6de79e630ea8f942c41:
  Author: Phil Elwell <phil@raspberrypi.org>
  Date:   Mon Nov 27 17:14:54 2017 +0000

    cgroup: Disable cgroup "memory" by default

    Some Raspberry Pis have limited RAM and most users won't use the
    cgroup memory support so it is disabled by default. Enable with:

        cgroup_enable=memory

    See: https://github.com/raspberrypi/linux/issues/1950

    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

Signed-off-by: Damian Wrobel <dwrobel@ertelnet.rybnik.pl>
---
 kernel/cgroup/cgroup.c | 38 --------------------------------------
 1 file changed, 38 deletions(-)

diff --git a/kernel/cgroup/cgroup.c b/kernel/cgroup/cgroup.c
index 4cb09fcdc16f6..de8b4fa1e1fd6 100644
--- a/kernel/cgroup/cgroup.c
+++ b/kernel/cgroup/cgroup.c
@@ -5824,9 +5824,6 @@ int __init cgroup_init_early(void)
 	return 0;
 }
 
-static u16 cgroup_enable_mask __initdata;
-static int __init cgroup_disable(char *str);
-
 /**
  * cgroup_init - cgroup initialization
  *
@@ -5865,12 +5862,6 @@ int __init cgroup_init(void)
 
 	mutex_unlock(&cgroup_mutex);
 
-	/*
-	 * Apply an implicit disable, knowing that an explicit enable will
-	 * prevent if from doing anything.
-	 */
-	cgroup_disable("memory");
-
 	for_each_subsys(ss, ssid) {
 		if (ss->early_init) {
 			struct cgroup_subsys_state *css =
@@ -6455,10 +6446,6 @@ static int __init cgroup_disable(char *str)
 			    strcmp(token, ss->legacy_name))
 				continue;
 
-			/* An explicit cgroup_enable overrides a disable */
-			if (cgroup_enable_mask & (1 << i))
-				continue;
-
 			static_branch_disable(cgroup_subsys_enabled_key[i]);
 			pr_info("Disabling %s control group subsystem\n",
 				ss->name);
@@ -6477,31 +6464,6 @@ static int __init cgroup_disable(char *str)
 }
 __setup("cgroup_disable=", cgroup_disable);
 
-static int __init cgroup_enable(char *str)
-{
-	struct cgroup_subsys *ss;
-	char *token;
-	int i;
-
-	while ((token = strsep(&str, ",")) != NULL) {
-		if (!*token)
-			continue;
-
-		for_each_subsys(ss, i) {
-			if (strcmp(token, ss->name) &&
-			    strcmp(token, ss->legacy_name))
-				continue;
-
-			cgroup_enable_mask |= 1 << i;
-			static_branch_enable(cgroup_subsys_enabled_key[i]);
-			pr_info("Enabling %s control group subsystem\n",
-				ss->name);
-		}
-	}
-	return 1;
-}
-__setup("cgroup_enable=", cgroup_enable);
-
 void __init __weak enable_debug_cgroup(void) { }
 
 static int __init enable_cgroup_debug(char *str)
-- 
2.35.3

